rust/private/clippy.bzl

```diff
@@ -26,38 +23,16 @@ load(
 )
 load("//rust/private:utils.bzl", "determine_output_hash", "find_cc_toolchain", "find_toolchain")
 
-_rust_extensions = [
-    "rs",
-]
-
-def _is_rust_target(srcs):
-    return any([src.extension in _rust_extensions for src in srcs])
-
-def _rust_sources(target, rule):
-    srcs = []
-    if "srcs" in dir(rule.attr):
-        srcs += [f for src in rule.attr.srcs for f in src.files.to_list()]
-    if "hdrs" in dir(rule.attr):
-        srcs += [f for hdr in rule.attr.hdrs for f in hdr.files.to_list()]
-    return [src for src in srcs if src.extension in _rust_extensions]
-
 def _clippy_aspect_impl(target, ctx):
     if rust_common.crate_info not in target:
         return []
-    rust_srcs = _rust_sources(target, ctx.rule)
 
     toolchain = find_toolchain(ctx)
     cc_toolchain, feature_configuration = find_cc_toolchain(ctx)
     crate_info = target[rust_common.crate_info]
+    rust_srcs = crate_info.srcs.to_list()
     crate_type = crate_info.type
-
-    if crate_info.is_test:
-        root = crate_info.root
-    else:
-        if rust_srcs == []:
-            # nothing to do
-            return []
-        root = crate_root_src(ctx.rule.attr, rust_srcs, crate_info.type)
+    root = crate_info.root
```

Comment:
nit: can you remove the `root` variable, and just use `crate_info.root` in the one case it is used?